web/src/beta/lib/reearth-ui/components/Selector/index.tsx

```diff
@@ -0,0 +1,252 @@
+import { FC, MouseEvent, useCallback, useEffect, useMemo, useRef, useState } from "react";
+
+import { styled, useTheme } from "@reearth/services/theme";
+
+import { Button } from "../Button";
+import { Icon } from "../Icon";
+import { Popup } from "../Popup";
+import { Typography } from "../Typography";
+
+export type SelectorProps = {
+  multiple?: boolean;
+  value?: string | string[];
+  options: { value: string; label?: string }[];
+  disabled?: boolean;
+  placeholder?: string;
+  onChange?: (value: string | string[]) => void;
+};
+
+export const Selector: FC<SelectorProps> = ({
+  multiple,
+  value,
+  options,
+  placeholder = "Please select",
+  disabled,
+  onChange,
+}) => {
+  const theme = useTheme();
+  const selectorRef = useRef<HTMLDivElement>(null);
+  const [selectedValue, setSelectedValue] = useState<string | string[] | undefined>(
+    value ?? (multiple ? [] : undefined),
+  );
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [selectorWidth, setSelectorWidth] = useState<number>();
+
+  const optionValues = useMemo(() => options, [options]);
+
+  useEffect(() => {
+    setSelectedValue(value ?? (multiple ? [] : undefined));
+  }, [value, multiple]);
+
+  const updateSelectorWidth = useCallback(() => {
+    if (selectorRef.current) {
+      const newWidth = selectorRef.current.clientWidth;
+      setSelectorWidth(newWidth);
+      console.log("Selector width updated:", newWidth);
+    }
+  }, []);
+
+  useEffect(() => {
+    updateSelectorWidth();
+    window.addEventListener("resize", updateSelectorWidth);
+    return () => {
+      window.removeEventListener("resize", updateSelectorWidth);
+    };
```

Comment:
```suggestion
    return window.addEventListener("resize", updateSelectorWidth);
```
addEventListener will return a remove function, we can use it as return function of useEffect directly.